libpromises/files_names.c

```diff
@@ -170,6 +170,40 @@ char *JoinSuffix(char *path, size_t path_size, const char *leaf)
     return path;
 }
 
+char *JoinPaths(char *path, size_t path_size, const char *leaf_path)
+/*
+Just like the JoinSuffix() above, but makes sure there's a FILE_SEPARATOR
+between @path and @leaf_path.
+*/
+{
+    size_t len = strlen(leaf_path);
+    size_t path_len = strlen(path);
+
+    if (Chop(path, path_size) == -1)
```

Comment:
Shouldn't this be `path_size-1` according to the comments in `Chop()`?

Reply:
Yeah, I believe it should be ``path_size-1``. I didn't check the code I copied over from ``JoinSuffix()``.